services/src/handlers/wms.rs

```diff
@@ -0,0 +1,91 @@
+use warp::{Filter, http::Response};
+use warp::reply::Reply;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+use crate::ogc::wms::request::{WMSRequest, GetCapabilities, GetMap};
+use geoengine_datatypes::raster::Raster2D;
+use geoengine_datatypes::operations::image::{Colorizer, ToPng};
+use crate::error::Error;
+use crate::workflows::registry::WorkflowRegistry;
+
+type WR<T> = Arc<RwLock<T>>;
+
+pub fn wms_handler<T: WorkflowRegistry>(workflow_registry: WR<T>) -> impl Filter<Extract=impl warp::Reply, Error=warp::Rejection> + Clone {
+    warp::get()
+        .and(warp::path!("wms"))
+        .and(warp::query::<WMSRequest>())
+        .and(warp::any().map(move || Arc::clone(&workflow_registry)))
+        .and_then(wms)
+}
+
+// TODO: move into handler once async closures are available?
+async fn wms<T: WorkflowRegistry>(request: WMSRequest, workflow_registry: WR<T>) -> Result<Box<dyn warp::Reply>, warp::Rejection> {
+    // TODO: authentication
+    // TODO: more useful error output than "invalid query string"
+    match request {
+        WMSRequest::GetCapabilities(request) => get_capabilities(&request),
+        WMSRequest::GetMap(request) => get_map(&request, &workflow_registry),
+        // TODO: support other requests
```

Comment:
I'm not sure if we should put a TODO on that because we went five years without needing other WMS requests.
I've seen `GetLegendGraphic` though ;-).

Reply:
ed0ac861634c7ccd93c379154e0cfe7bf09bf9fc